### PR TITLE
fix: keep the ttl bit when dense_set grows with expiring items

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: src/core/string_set_test.cc [hunk 1]
```diff
@@ -135,27 +135,30 @@ static string random_string(mt19937& rand, unsigned len) {
 
 TEST_F(StringSetTest, Resizing) {
   constexpr size_t num_strs = 4096;
-  vector<string> strs;
+  unordered_set<string> strs;
```
Comment (BorysTheDev) [inline src/core/string_set_test.cc:138]: why?

Comment (BorysTheDev) [reply src/core/string_set_test.cc:138]: If you want to make it unique, std::unique is a better option

**Review by BorysTheDev — APPROVED** [review]

**Review by BorysTheDev — DISMISSED** [review]

File: src/core/string_set_test.cc [hunk 2]
```diff
@@ -121,6 +121,58 @@ TEST_F(StringSetTest, StandardAddErase) {
   EXPECT_TRUE(ss_->Erase("AAAAAAAAAAAAAAA@"));
 }
 
+TEST_F(StringSetTest, DisplacedBug) {
+  string_view vals[] = {"imY", "OVl", "NhH", "BCe", "YDL", "lpb",
+                        "nhF", "xod", "zYR", "PSa", "hce", "cTR"};
+  ss_->AddMany(absl::MakeSpan(vals), UINT32_MAX);
+
+  ss_->Add("fIc");
```
Comment (kostasrim) [inline src/core/string_set_test.cc:129]: :scream:

Comment (kostasrim) [reply src/core/string_set_test.cc:129]: (that builder pattern would be :100: here -- it's fine we don't really needed just mentioning this)

File: src/core/string_set_test.cc (hunk 1 above)
Comment (romange) [reply src/core/string_set_test.cc:190]: it's not better. it assumes the vector is sorted. this is simpler.

**Pull request merged**